### PR TITLE
metric[engine]:Replace framework metric

### LINKED ISSUE #5525: replace framework prometric metric with multi-projects/tasks metric

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #5525 

### What is changed and how it works?
replace prometric metric with multi-projects/tasks metric

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/pkg/promutil/util.go [hunk 1]
```diff
@@ -72,3 +72,10 @@ func NewFactory4Worker(info tenant.ProjectInfo, jobType libModel.JobType, jobID
 func NewFactory4Framework() Factory {
 	return NewFactory4FrameworkImpl(globalMetricRegistry)
 }
+
+// UnregisterWorkerMetrics unregisters all metric of workerID
```
Comment (amyangfei) [inline engine/pkg/promutil/util.go:76]:
```suggestion
// UnregisterWorkerMetrics unregisters all metrics of workerID
```

File: engine/pkg/promutil/util.go [hunk 2]
```diff
@@ -72,3 +72,10 @@ func NewFactory4Worker(info tenant.ProjectInfo, jobType libModel.JobType, jobID
...
+// IF 'worker' is a job master, use job id as workerID
+// IF 'worker' is a worker, use worke id as workerID
```
Comment (amyangfei) [inline engine/pkg/promutil/util.go:78]:
```suggestion
// IF 'worker' is a worker, use worker id as workerID
```

File: engine/lib/base_jobmaster.go [hunk 3]
```diff
@@ -180,13 +181,14 @@ func (d *DefaultBaseJobMaster) GetWorkers() map[libModel.WorkerID]WorkerHandle {
...
-		return errors.Trace(err)
+	err := d.impl.CloseImpl(ctx)
+	if err != nil {
+		log.L().Error("Failed to close JobMasterImpl", zap.Error(err))
 	}
 
 	d.master.doClose()
 	d.worker.doClose()
-	return nil
+	return errors.Trace(err)
```
Comment (amyangfei) [inline engine/lib/base_jobmaster.go:191]: Please add a comment here about why we don't return err in L186 before `d.worker.doClose` is called.

File: engine/lib/master.go [hunk 4]
```diff
@@ -435,16 +435,18 @@ func (m *DefaultBaseMaster) doClose() {
...
-	if err := m.Impl.CloseImpl(ctx); err != nil {
-		return errors.Trace(err)
+	err := m.Impl.CloseImpl(ctx)
+	if err != nil {
+		log.L().Error("Failed to close MasterImpl", zap.Error(err))
 	}
 
 	m.doClose()
-	return nil
+	return errors.Trace(err)
```
Comment (amyangfei) [inline engine/lib/master.go:449]: ditto

File: engine/lib/worker.go [hunk 5]
```diff
@@ -350,17 +350,18 @@ func (w *DefaultBaseWorker) doClose() {
...
-	if err := w.Impl.CloseImpl(ctx); err != nil {
+	err := w.Impl.CloseImpl(ctx)
+	if err != nil {
 		log.L().Error("Failed to close WorkerImpl", zap.Error(err))
-		return errors.Trace(err)
 	}
 
 	w.doClose()
-	return nil
+	return errors.Trace(err)
```
Comment (amyangfei) [inline engine/lib/worker.go:364]: ditto

Comment (maxshuang) [reply engine/lib/worker.go]: Done

File: engine/lib/master.go (hunk 4 above)
Comment (maxshuang) [reply engine/lib/master.go]: Done

File: engine/lib/base_jobmaster.go (hunk 3 above)
Comment (maxshuang) [reply engine/lib/base_jobmaster.go]: Done

**Review by amyangfei — APPROVED** [review]

**Review by liuzix — APPROVED** [review]

**Pull request merged**